### PR TITLE
Debian Bullseye

### PR DESCRIPTION
**Pull request by J0WI**

### CODE REVIEW

File: library.pl [hunk 1]
```diff
@@ -7,8 +7,9 @@
 my %arches = (
 
   # https://github.com/docker-library/official-images/blob/master/library/debian
-  buster  => 'amd64, arm32v7, arm64v8, i386, ppc64le, s390x',
-  stretch => 'amd64, arm32v7, arm64v8, i386',
+  bullseye => 'amd64, arm32v5, arm32v7, arm64v8, i386, mips64le, ppc64le, s390x',
+  buster => 'amd64, arm32v5, arm32v7, arm64v8, i386, mips64le, ppc64le, s390x',
+  stretch => 'amd64, arm32v5, arm32v7, arm64v8, i386',
```
[the remark this answers is not in the document]
Comment (zakame) [reply library.pl:12]: Got it, thanks for clarifying!

**Review by level420 — CHANGES_REQUESTED** [review]

File: library.pl (hunk 1 above)
[the remark this answers is not in the document]
Comment (J0WI) [reply library.pl]: No.

[the remark this answers is not in the document]
Comment (zakame) [reply library.pl]: Good point, thanks @level420 @J0WI 🙏 !

**Review by level420 — APPROVED** [review]
👍 @zakame @J0WI

**Comment by level420** [comment]
@zakame do you have a eta when the bullseye image will be available on docker hub?

**Review by zakame — APPROVED** [review]
LGTM 👍 

<details>

<summary>Build of perl:5.34-main-bullseye, click to expand:</summary>

```
[zakame:~/src/docker-perl/5.034.000-main-bullseye] bullseye ± docker build -t perl:5.34-main-bullseye .       
[+] Building 818.7s (10/10) FINISHED                                                                                                                                                                                                          
 => [internal] load build definition from Dockerfile                                                                                                                                                                                     0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                      0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                                                                                          0.0s
 => [internal] load metadata for docker.io/library/buildpack-deps:bullseye                                                                                                                                                               5.2s
 => [auth] library/buildpack-deps:pull token for registry-1.docker.io                                                                                                                                                                    0.0s
 => [internal] load build context                                                                                                                                                                                                        0.0s
 => => transferring context: 41B                                                                                                                                                                                                         0.0s
 => [1/5] FROM docker.io/library/buildpack-deps:bullseye@sha256:1062c3cabe1b62d7e64609e1ebe30406b9d802684758a01fd386bbc60ea6b1aa                                                                                                       153.2s
 => => resolve docker.io/library/buildpack-deps:bullseye@sha256:1062c3cabe1b62d7e64609e1ebe30406b9d802684758a01fd386bbc60ea6b1aa                                                                                                         0.0s
 => => sha256:889b1f128be8ebd64f787c46418ffe34ce1096d1ccd6938924d7397713720758 5.14MB / 5.14MB                                                                                                                                          14.8s
 => => sha256:c338d42ad921985fb8ebb6e2c48d381f6e03a91535eeffce7f08084b3dfbfdf4 10.66MB / 10.66MB                                                                                                                                         9.8s
 => => sha256:1062c3cabe1b62d7e64609e1ebe30406b9d802684758a01fd386bbc60ea6b1aa 1.86kB / 1.86kB                                                                                                                                           0.0s
 => => sha256:cce82fb6cf801fc490ffefa5faa4ce168ba7e5b6002ad8de9706aa3ba06b243b 1.38kB / 1.38kB                                                                                                                                           0.0s
 => => sha256:aed3f8ee39bdf4b2b27153234abddad555faf343c1ead2b65b98ad77b20567b9 4.41kB / 4.41kB                                                                                                                                           0.0s
 => => sha256:1c47a423366578e5ce665d03788914bf0459485a627a27896fa9c5663ce55cdf 53.60MB / 53.60MB                                                                                                                                        73.1s
 => => sha256:e27d3d9a061af78e3fcbefe5dbbe718db380687319e5ba8a7c9fd7ba55d16cc3 54.67MB / 54.67MB                                                                                                                                        46.6s
 => => sha256:56467c8996a237c29310630374ed68e9883e8e7117197de0272f3275c9c6ce75 189.39MB / 189.39MB                                                                                                                                     148.3s
 => => extracting sha256:1c47a423366578e5ce665d03788914bf0459485a627a27896fa9c5663ce55cdf                                                                                                                                                1.3s
 => => extracting sha256:889b1f128be8ebd64f787c46418ffe34ce1096d1ccd6938924d7397713720758                                                                                                                                                0.1s
 => => extracting sha256:c338d42ad921985fb8ebb6e2c48d381f6e03a91535eeffce7f08084b3dfbfdf4                                                                                                                                                0.2s
 => => extracting sha256:e27d3d9a061af78e3fcbefe5dbbe718db380687319e5ba8a7c9fd7ba55d16cc3                                                                                                                                                1.5s
 => => extracting sha256:56467c8996a237c29310630374ed68e9883e8e7117197de0272f3275c9c6ce75                                                                                                                                                4.6s
 => [2/5] COPY *.patch /usr/src/perl/                                                                                                                                                                                                    0.1s
 => [3/5] WORKDIR /usr/src/perl                                                                                                                                                                                                          0.0s
 => [4/5] RUN true     && curl -SL https://www.cpan.org/src/5.0/perl-5.34.0.tar.xz -o perl-5.34.0.tar.xz     && echo '82c2e5e5c71b0e10487a80d79140469ab1f8056349ca8545140a224dbbed7ded *perl-5.34.0.tar.xz' | sha256sum -c -     && t  240.9s
 => exporting to image                                                                                                                                                                                                                   0.2s
 => => exporting layers                                                                                                                                                                                                                  0.2s
 => => writing image sha256:d4f1961d4750f006792b4346dcce26cd49952be7109c94f7aab7000679c3d39c                                                                                                                                             0.0s
 => => naming to docker.io/library/perl:5.34-main-bullseye                                                                                                                                                                               0.0s
                                                                                                                                                                                                                                              
```

</details>

**Comment by zakame** [comment]
@level420 @J0WI Thanks for the PR!  Sorry for the delay - aside from the CI checks above, had to test these on my machines/workloads as well.

https://doi-janky.infosiftr.net/job/multiarch/view/images/view/perl/ will probably start building these (at least the `multiarch` ones) as soon as they're synced with the latest merge, so it should also push to the hub once ready - I think @tianon would know better.

**Comment by zakame** [comment]
Also needs corresponding https://github.com/docker-library/official-images PR to add the `bullseye` tags

**Comment by level420** [comment]
@zakame great thank you!
99.9% kudos to @J0WI, 0.01% to me 😃 
Landed on docker hub:
https://hub.docker.com/_/perl?tab=tags&page=1&name=bullseye

**Pull request merged**